### PR TITLE
feat(linter): add `no-unnecessary-docblock-tag` rule

### PR DESCRIPTION
**Pull request by MidnightDesign**
> [!NOTE]
> This PR was written by AI (Claude Code). The full session transcript will be posted as a GitHub Gist.
> I ([@MidnightDesign](https://github.com/MidnightDesign)) don't have too much experience with Rust, so I confirmed manually that:
> 1. The rule descriptions are accurate and readable
> 2. The tests cover all edge cases I could think of
> 3. The code doesn't contain any Rust red flags like `.unwrap()`

## Summary

- Detects `@param` and `@return` docblock tags that merely repeat the type already present in the function/method signature without adding any description.
- Tags that provide more specific types (e.g., `list<string>` for an `array` parameter) or include descriptions are **not** flagged.
- Walks functions, methods in classes/interfaces/traits/enums, and namespaced code.

Disabled by default (pedantic), with `Level::Help`.

## Test plan

- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --workspace --all-features -- -D warnings` passes
- [x] `cargo test --workspace --locked` passes
- [x] 8 unit tests covering success and failure cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by MidnightDesign** [comment]
Full session transcript: https://gist.github.com/MidnightDesign/2b964b4931a73de737dd18370e1d873a

File: crates/linter/src/rule/consistency/ambiguous_constant_access.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
+use indoc::indoc;
```
Comment (azjezz) [inline crates/linter/src/rule/consistency/ambiguous_constant_access.rs]: I think you pushed the wrong code here.. this is same as #1494

Comment (MidnightDesign) [reply crates/linter/src/rule/consistency/ambiguous_constant_access.rs]: Sorry about that. Should be the right code now.

**Comment by MidnightDesign** [comment]
We basically have the same issue as in https://github.com/carthage-software/mago/pull/1496#discussion_r2994289292. To get to the docblock, we need the `Program`, so currently the only option seems to be to have `Program` be the target and iterate through the entire codebase.

The only way to solve this is probably to reference doc blocks from their corresponding nodes, so you can just target those nodes in the rule and grab the doc block from there.

**Comment by azjezz** [comment]
Same as #1496 - needs to wait for new docblock parser.
